packages/compass/src/main/handle-uncaught-exception.ts

```diff
@@ -0,0 +1,54 @@
+import { app, dialog, clipboard } from 'electron';
+import cleanStack from 'clean-stack';
+import ensureError from 'ensure-error';
+import COMPASS_ICON from './icon';
+
+function handleUncaughtException(err: Error): void {
+  // eslint-disable-next-line no-console
+  console.error('handling uncaughtException', err);
+  err = ensureError(err);
+  const stack = cleanStack(err.stack || '');
+
+  const detail = `${app.getName()} version ${app.getVersion()}\nStacktrace:\n${stack}`;
+  const message = `${app.getName()} has encountered an unexpected error`;
+
+  // eslint-disable-next-line no-console
+  console.error(`${message}: ${detail}`);
+
+  const showErrorMessageBox = async () => {
+    const { response } = await dialog.showMessageBox({
+      type: 'error',
+      buttons: [
+        'OK',
+        process.platform === 'darwin' ? 'Copy Error' : 'Copy error',
+      ],
+      icon: COMPASS_ICON,
+      defaultId: 0,
+      noLink: true,
+      message: message,
+      detail: detail,
+    });
+
+    /**
+     * TODO (@imlucas) Copy diagnostics to clipboard?
```

Comment:
Maybe we should use this opportunity to remove TODOs from years past that we're not going to do unless a ticket to do it comes up.